base_search_mail_content/models/mail_thread.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# © 2016-17 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# © 2016 Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from lxml import etree
+from odoo.osv import expression
+from odoo.osv.orm import setup_modifiers
+
+
+class MailThread(models.AbstractModel):
+
+    _inherit = 'mail.thread'
+
+    def _search_message_content(self, operator, value):
+
+        model_domain = [('model', '=', self._name)]
+        if operator not in expression.NEGATIVE_TERM_OPERATORS:
+            model_domain += ["|"] * 4
+        model_domain += [
+            ('record_name', operator, value),
+            ('subject', operator, value),
+            ('body', operator, value),
+            ('email_from', operator, value),
+            ('reply_to', operator, value)
+        ]
+        recs = self.env['mail.message'].search(model_domain)
+        return [('id', 'in', recs.mapped('res_id'))]
+
+    message_content = fields.Text(
+        string='Message Content',
+        help='Message content, to be used only in searches',
+        compute=lambda self: False,
+        search='_search_message_content')
+
+
+class BaseModel(object):
```

Comment:
I can't seem to do quick search of message contents. Most likely this should inherit from `models.AbstractModel` instead, although I think you already fixed this before... 🤔

Reply:
Postgresql decides when is best to use trigram indexes. You will need to install this module into a DB with a rather large number of emails. 

To check if the trigram indexes are being used, you must:

1. start odoo with --log-level=debug_sql
2. Do the search that you want to analyze - e.g. search for emails from the leads. 
3. Copy the query and run in command line EXPLAIN ANALYZE (my_query)

EXPLAIN ANALYZE (SELECT "mail_message".id FROM "mail_message" WHERE (("mail_message"."model" = 'res.partner')  AND  ((((("mail_message"."record_name"::text ilike 'dear')  OR  ("mail_message"."subject"::text ilike 'dear'))  OR  ("mail_message"."body"::text ilike 'dear'))  OR  ("mail_message"."email_from"::text ilike 'dear'))  OR  ("mail_message"."reply_to"::text ilike 'dear'))) ORDER BY "mail_message"."id" DESC);

This will tell you if trigram indexes are being used.

Reply:
![what?](https://media.giphy.com/media/yZ2FSn86bf2co/giphy.gif)